\documentclass[10pt]{article}
\usepackage[margin=1in]{geometry}
\usepackage{fancyhdr}
\usepackage{wrapfig}
\usepackage{amsmath}
\usepackage{enumerate}
\usepackage{placeins}
\usepackage{color}
\usepackage{hyperref}
\hypersetup{
    colorlinks,
    citecolor=blue,
    filecolor=blue,
    linkcolor=blue,
    urlcolor=blue
}
\usepackage{array}
\newcolumntype{L}{>{\arraybackslash}m{3cm}}
\setlength{\parskip}{3mm}
\pagestyle{fancy}
\lhead{Final Project \hspace{3mm}  BCB570  \hspace{3mm} Spring \hspace{3mm} Due May 6}
\rhead{Zebulun Arendsee}

\begin{document}

\section{Abstract}
\label{sec:abstract}

It is now well accepted that novel genes can arise de novo from non-genic
material. However, little is known about the regulational circuitry of these
young genes. A few studies have searched for regulatory motifs in the orphan
promoters, but prediction of \textit{functional} binding sites, often having
only a few bits of information, is not reliable. I will instead 1) use RNA-seq
data to predict transcription factors (TF) that regulate each gene and 2)
compare the results to sequence-based binding site predictions.

There are several ways an orphan may acquire existing transcriptional codes.
They may exploit the promoter of an established gene as a secondary open
reading frame; fall under the paradigm of a transposon that inserts upstream;
inherit the full regulatory context of a host that has lost its primary gene;
or emerge as a translational byproduct of a functional long non-coding RNA.
However, there is also evidence that orphans can arise completely de novo.
Though it may at first seem unlikely that all the transcriptional and
translational prerequisites would assemble by chance, given the small amount of
information required for most signals, and given the number of chances a
population of genomes across thousands of generations have, chance emergence of
de novo orphans may not be so uncommon.

Orphan genes of truly de novo origin should be directly regulated by only a few
transcription factors with simple binding sites. Alternatively, orphans could
hijack the transcriptional 

\section{Background}
\label{sec:background}

\begin{enumerate}
  \item Take all TAIR10 models
  \item Merge genes with identical coding sequences
  \item Build kallisto index with these genes
  \item Select set of interesting Arabidopsis studies
  \begin{itemize}
      \item want wide variety of conditions
      \item use SRAdb to get list of runids
      \item for each, fetch | fastq-dump | kallisto
  \end{itemize}
  \item In Sleuth, normalize all the studies merge into final expression table
\end{enumerate}

\begin{enumerate}
  \item Do the orphans cluster together in expression?
  \item If so, do they cluster more than I would expect?
  \item Take the largest or most interesting cluster
  \begin{enumerate}
    \item  Do they share more interactions than I would expect? If so, this is
       evidence that the expression clustering has some basis in protein
       function, rather than chance.
    \item Do they share a common, simple promoter motif?
    \item Do they share a common upstream promoter?
  \end{enumerate}
\end{enumerate}

Naive hypothesis:

If the genes are co-expressed they must be part of the same functional
pathway. Expression is driven by selection.

Other possibilities:

They share a common origin. Expression is neutral. The co-expression
predates any shared functionality.

Story 1: A transposon proliferation provided the promoter machinery that
animated the non-genic precursor of the orphan.

Story 2: If the promoter is completely de novo, the most likely regulatory
motifs to have arisen are the shortest, most simple ones.

Perhaps, because they are coexpressed originally, and since they therefore
have the chance to interact, selection will work to enhance the
similarities, thus drawing the cluster together.

I want to know about the promoters motifs that are being used by the
orphan genes. However, predicting functional binding sites de novo is
extraordinarily difficult. So I will start by searching for transcription
factors that are correlated with the particular orphan genes.

I will treat each orphan as if it is the sole output of the transcription
factor network. Then I will build a correlation network including the
transcription factors and the orphan. I assume there are two types of genes:
transcription factors and non-transcription factors. In a directed graph, the
transcription factors will be internal nodes, and the non-transcription factors
are leafs. Thus, non-transcription factors are all unconnected and independent.
This allows me to model the transcription factors and a single gene, ignoring
all others. By reducing the search space so greatly, I can use more
computationally intensive tools than would otherwise be feasible.

Of course, my assumption that non-transcription factors have no effect on the
transcription of other genes is not entirely true. A protein can, for example,
catalyze the production of a metabolite that activates a transcriptional
repressor of another gene.

Do the orphan genes cluster in their own little networks? If so, are these
functional clusters or artifacts of common origin? For example, were they
all animated by insertions by the same type of telomere?

Orphan problems:

    Generally GO is not an option. There are only a few orphans that even have
    GO terms.

    Generally, evolutionary methods are not available.

    There is a few added layer of uncertainty: the orphans may not actually
    express a functional (whatever that means) protein product. They may be
    noise.

Unfortunately, the exprimental structures of RNA-seq data are not encoded in
the SRA metadata.

\section{Methods}
\label{sec:methods}

\subsection{Collecting RNA-seq data}
\label{sub:collecting_rna_seq_data}

\begin{enumerate}
  \item Arabidopsis thaliana RNA-seq: take maybe 3 studies. Run through my
   RNA-seq pipeline using Kallisto. Process with Sleuth. Include the
   bootstrapping procedure.
  \item Use snakefile to organize run
  \item Compare to interaction network data, using IntAct
  \item Compare to promoter motif set similarity. How to do this? Retrieve
   putative set of discrete regulatory elements (one exists for
   Arabidopsis), Draw edges between genes if there share a common element.
   Weights are relative to the number of shared elements. This is not the
   only way to do it, but it may work.
  \item Compare genomic distance network.
\end{enumerate}

From the selected studies, I removed any runs that

 1. Where not contentional transcriptomic (for example, bisulfite sequencing)
 1. Where not from Col-0 *A. thaliana*

\begin{table}[htpb]
    \centering
    \caption{Studies used to build the coexpression networks. D means control,
        BR brassinosteroid, and GA gibberillin.}
    \label{tab:sra_table}
    \begin{tabular}{l | L | L | L}
        accession & description & layout & notes \\
        \hline
        ERP004034 & floral development & (inflorecence meristem, floral meristem, flower) & {} \\
        \hline
        ERP005391 & somatic embryo development & (three stages) & {} \\
        \hline
        SRP003864 & copper defficiency & (normal Cu, low Cu) & Deleted spl7 mutants \\
        \hline
        SRP007113 & female gametophyte & {} & \\
        \hline
        SRP009369 & root hair & {} & {} \\
        \hline
        SRP010642 & hormone treatments & (D, BR-, GA-, BR-GA-) X 2 & Deleted bzr1 mutants \\
           %PAC-PPZ negative for both the GA and BR hormones
           %PAC-GA  negative for BR
           %PAC     negative for GA
        \hline
        SRP018404 & root nitrate response & (normal, high) & Deleted the sRNA runs \\
        \hline
        SRP027256 & drought & {} & Deleted the GI mutants \\
        \hline
        SRP034715 & mildew pathoge & (virulent, avirulent) X 3 & Deleted the water control and the avirulent Emoy2 infected runs \\
        \hline
        SRP035269 & seed development & (3 dev stages) X 3 & Deleted lines CL37 and PDAT\_7, keeping only Fae  \\
        \hline
        SRP044814 & Fe stress & {} & {} \\
        \hline
        SRP063017 & Nematode infection & {} & Deleted cytokine hypersensitive type-A lines \\
        \hline
    \end{tabular}
\end{table}
% Honorable mentions: 1) SRP051665 is interesting in that it attempted to study
% the translatome by extracting mRNAs from polyribosomes. 2) SRP032274 also uses
% PPZ to stop BR hormongering. SRP032366 heat stress, 38C for 15 minutes, deleted
% for lack of biological replicates.

\subsection{Quantify transcript expression}
\label{sub:quantify_transcript_expression}

For expression quantification we are using a new tool, Kallisto, which is
a few orders of magnitude faster than the conventional Bowtie and
Cufflinks approach. Kallisto is so fast, and so easily parallizeable, that
the bottleneck in the pipeline is data retrieval.

Kallisto is a very fast tool for quantitative analysis of transcriptome
expression \cite{bray_near-optimal_2015}.

at.fa - CDS for arabidopsis models (retrieved from the Kallisto website, and
they got it from Ensembl)

Then Sleuth

\subsection{Select transcription factor set}
\label{sub:select_transcription_factor_set}

bindingsite-data.tbl - experimentally confirmed [TF-binding
sites](http://arabidopsis.med.ohio-state.edu/AtcisDB/bindingsites.html) from
Agris.

\subsection{Orphan annotations}
\label{sub:orphan_annotations}

orphans - an in-house list of orphan TAIR10 model ids.

\subsection{Predict direct regulators}
\label{sub:predict_direct_regulators}

There are several possible approaches I could use here.

The easiest possibility would be to use ARACNE to trim indirect interactions.
Any links remaining between the orphan and a transcription factor would be
assumed to be causal. Although it is possible that the orphan itself is a
transcription factor.

A possibly more powerful method is a form of intervention calculus that builds
a DAG from observed data \cite{maathuis} and the pcalg R library.

\subsection{Compare expression predictions to motif predictions}
\label{sub:compare_expression_predictions_to_motif_predictions}

\subsection{Build full expression matrix}
\label{sub:build_full_expression_matrix}

\section{Conclusions}
\label{sec:conclustions}

\bibliographystyle{plain}
\bibliography{report}

\end{document}
