\documentclass[10pt]{article}
\usepackage[margin=1in]{geometry}
\usepackage{fancyhdr}
\usepackage{wrapfig}
\usepackage{amsmath}
\usepackage{enumerate}
\usepackage{placeins}
\setlength{\parskip}{3mm}
\pagestyle{fancy}
\lhead{Final Project \hspace{3mm}  BCB570  \hspace{3mm} Spring \hspace{3mm} Due May 6}
\rhead{Zebulun Arendsee}

\begin{document}

\section{Introduction}
\label{sec:introduction}

I want to know about the promoters motifs that are being used by the
orphan genes. However, predicting functional binding sites de novo is
extraordinarily difficult. So I will start by searching for transcription
factors that are correlated with the particular orphan genes.

I will treat each orphan as if it is the sole output of the transcription
factor network. Then I will build a correlation network including the
transcription factors and the orphan. I assume there are two types of genes:
transcription factors and non-transcription factors. In a directed graph, the
transcription factors will be internal nodes, and the non-transcription factors
are leafs. Thus, non-transcription factors are all unconnected and independent.
This allows me to model the transcription factors and a single gene, ignoring
all others. By reducing the search space so greatly, I can use more
computationally intensive tools than would otherwise be feasible.

Of course, my assumption that non-transcription factors have no effect on the
transcription of other genes is not entirely true. A protein can, for example,
catalyze the production of a metabolite that activates a transcriptional
repressor of another gene.

Do the orphan genes cluster in their own little networks? If so, are these
functional clusters or artifacts of common origin? For example, were they
all animated by insertions by the same type of telomere?

1. Arabidopsis thaliana RNA-seq: take maybe 3 studies. Run through my
   RNA-seq pipeline using Kallisto. Process with Sleuth. Include the
   bootstrapping procedure.

1. Use snakefile to organize run

1. Compare to interaction network data, using IntAct

1. Compare to promoter motif set similarity. How to do this? Retrieve
   putative set of discrete regulatory elements (one exists for
   Arabidopsis), Draw edges between genes if there share a common element.
   Weights are relative to the number of shared elements. This is not the
   only way to do it, but it may work.

1. Compare genomic distance network.

Questions:

1. Do the orphans cluster together in expression?

1. If so, do they cluster more than I would expect?

1. Take the largest or most interesting cluster

    1. Do they share more interactions than I would expect? If so, this is
       evidence that the expression clustering has some basis in protein
       function, rather than chance.
    
    1. Do they share a common, simple promoter motif?

    1. Do they share a common upstream promoter?

Naive hypothesis:

If the genes are co-expressed they must be part of the same functional
pathway. Expression is driven by selection.

Other possibilities:

They share a common origin. Expression is neutral. The co-expression
predates any shared functionality.

Story 1: A transposon proliferation provided the promoter machinery that
animated the non-genic precursor of the orphan.

Story 2: If the promoter is completely de novo, the most likely regulatory
motifs to have arisen are the shortest, most simple ones.

Perhaps, because they are coexpressed originally, and since they therefore
have the chance to interact, selection will work to enhance the
similarities, thus drawing the cluster together.


Orphan problems:

    Generally GO is not an option. There are only a few orphans that even have
    GO terms.

    Generally, evolutionary methods are not available.

    There is a few added layer of uncertainty: the orphans may not actually
    express a functional (whatever that means) protein product. They may be
    noise.

\end{document}
